eventmesh-connectors/eventmesh-connector-chatgpt/src/main/java/org/apache/eventmesh/connector/chatgpt/source/connector/ChatGPTSourceConnector.java

```diff
@@ -0,0 +1,232 @@
+    private String parsePromptTemplateStr;
+    private ChatHandler chatHandler;
+    private ParseHandler parseHandler;
+
+    @Override
+    public Class<? extends Config> configClass() {
+        return ChatGPTSourceConfig.class;
+    }
+
+    @Override
+    public void init(Config config) {
+        this.sourceConfig = (ChatGPTSourceConfig) config;
+        doInit();
+    }
+
+    @Override
+    public void init(ConnectorContext connectorContext) {
+        SourceConnectorContext sourceConnectorContext = (SourceConnectorContext) connectorContext;
+        this.sourceConfig = (ChatGPTSourceConfig) sourceConnectorContext.getSourceConfig();
+        doInit();
+    }
+
+    public void initParsePrompt() {
+        String parsePromptFileName = sourceConfig.getConnectorConfig().getParsePromptFileName();
+        URL resource = Thread.currentThread().getContextClassLoader().getResource(parsePromptFileName);
+        if (resource == null) {
+            log.warn("cannot find prompt file {} in resources", parsePromptFileName);
+            return;
+        }
+        String filePath = resource.getPath();
+        try (BufferedReader br = new BufferedReader(new FileReader(filePath))) {
+            StringBuilder builder = new StringBuilder();
+            String line;
+            while ((line = br.readLine()) != null) {
+                if (!line.startsWith("#") && StringUtils.isNotBlank(line)) {
+                    builder.append(line).append("\n");
+                }
+            }
+            this.parsePromptTemplateStr = builder.toString();
+        } catch (IOException e) {
+            throw new IllegalStateException("Unable to read file", e);
+        }
+    }
+
+
+    @SuppressWarnings("checkstyle:WhitespaceAround")
+    private void doInit() {
+        initParsePrompt();
+        this.openaiManager = new OpenaiManager(sourceConfig);
+        this.chatHandler = new ChatHandler(this.openaiManager);
+        if (StringUtils.isNotEmpty(parsePromptTemplateStr)) {
+            this.parseHandler = new ParseHandler(openaiManager, parsePromptTemplateStr);
+        }
+        this.queue = new LinkedBlockingQueue<>(1024);
+        final Vertx vertx = Vertx.vertx();
+        final Router router = Router.router(vertx);
+        router.route().path(this.sourceConfig.connectorConfig.getPath()).method(HttpMethod.POST).handler(BodyHandler.create()).handler(ctx -> {
+            try {
+                RequestBody body = ctx.body();
+                ChatGPTRequestDTO bodyObject = body.asPojo(ChatGPTRequestDTO.class);
+                validateRequestDTO(bodyObject);
+                handleRequest(bodyObject, ctx);
+            } catch (Exception e) {
+                handleError(e, ctx);
+            }
+        });
+        this.server = vertx.createHttpServer(new HttpServerOptions().setPort(this.sourceConfig.connectorConfig.getPort())
+            .setIdleTimeout(this.sourceConfig.connectorConfig.getIdleTimeout())).requestHandler(router);
```

Comment:
How would Vertx handle it when `idleTimeout` is 0?

Reply:
The same problem as openaiTimeout will also occur. I added the processing logic and rechecked the places where default values ​​need to be added.